### PR TITLE
Allow to set/edit note on shares

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This allows to add notes on sharees with NC14+, same like in web UI.

 

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -822,5 +822,10 @@
     <string name="single_sign_on_request_token" formatted="true">Allow %1$s to access your Nextcloud account %2$s?</string>
     <string name="permission_deny">Deny</string>
     <string name="permission_allow">Allow</string>
+    <string name="share_send_note">Note to recipient</string>
+    <string name="note_confirm">Send</string>
+    <string name="send_note">Send note to recipient</string>
+    <string name="note_could_not_sent">Could not send note!</string>
```
Comment (AndyScherzinger) [inline src/main/res/values/strings.xml]: I'd drop the exclamation mark. No house on fire, no people in danger ;)

File: src/main/res/menu/file_detail_sharing_link_menu.xml [hunk 2]
```diff
@@ -44,4 +44,10 @@
         android:title="@string/share_via_link_send_link_label"
         app:showAsAction="never" />
 
+    <item
```
Comment (AndyScherzinger) [inline src/main/res/menu/file_detail_sharing_link_menu.xml]: So share note is available for share-links too?

**Comment by AndyScherzinger** [comment]
code-wise fine with me

> I set EditText to 3 lines, but I am not sure if this looks good UI-wise.

I think it looks okay, don't have a better solution at hand at the moment to be honest.

:exclamation: 2 new lint issues though

File: src/main/res/values/strings.xml (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/res/values/strings.xml]: :rofl:

File: src/main/res/menu/file_detail_sharing_link_menu.xml (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/res/menu/file_detail_sharing_link_menu.xml]: Yes, sir :-D
It will then appear on top right as a little text bubble

**Comment by tobiasKaminsky** [comment]
>  2 new lint issues though

1 auto fill: I'll address them all at once in a later PR
1 inflate with null as root view: this is ok for dialogs as they have no root view.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky how can I test this? I run 14.0.1.1 as the server part and I can create a note but I don't see the note anywhere, neither on the server, nor on the Android client? :(

**Comment by tobiasKaminsky** [comment]
On server you should see it when opening the share sidebar.
On android you currently cannot see it. I will create a different PR for this, but I first need some info on how to retrieve the note.

**Comment by AndyScherzinger** [comment]
:+1: fine by me then :)

**Comment by mario** [comment]
👍

**Pull request merged**